themes/wds_headless/inc/options-page.php

```diff
@@ -0,0 +1,280 @@
+}
+
+/**
+ * Retrieves an associative array of constants defined in PHP.
+ *
+ * @access private
+ *
+ * @return array The associative array of PHP constants.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function _wds_headless_get_defined_constants() : array {
+	return \get_defined_constants( true );
+}
+
+/**
+ * Verifies that all of the WDS Headless constants are defined. If all of them are defined,
+ * this function will return true; otherwise, it will return false ignoring any that are
+ * defined.
+ *
+ * @access private
+ *
+ * @return bool True if all options are set in the wp-config.php file.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function _wds_headless_has_defined_all_constants() {
+	return (
+		_wds_headless_check_wp_config( 'HEADLESS_FRONTEND_URL', 'wds-headless-frontend-url' ) &&
+		_wds_headless_check_wp_config( 'WORDPRESS_PREVIEW_SECRET', 'wds-headless-preview-secret' ) &&
+		_wds_headless_check_wp_config( 'GRAPHQL_JWT_AUTH_SECRET_KEY', 'wds-headless-jwt-auth-key' )
+	);
+}
+
+/**
+ * Updates the options table with the specified options page with the value of the constant
+ * set in wp-config.php.
+ *
+ * @access private
+ *
+ * @param string $wds_headless_constant The PHP constant to check before saving data from the form.
+ * @param string $option_name           The name of the option to serialize the value of the constant.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function _wds_headless_update_options( string $wds_headless_constant, string $option_name ) {
+	$constants = _wds_headless_get_defined_constants();
+	update_option( $option_name, $constants['user'][ strtoupper( $wds_headless_constant ) ] );
+}
+
+
+/*
+ * Option Page Settings.
+ * These functions use the WordPress API and the utility functions to populate the options
+ * and to support user interaction.
+ */
+
+add_action( 'after_setup_theme', 'wds_headless_read_wp_config' );
+/**
+ * This determines if the constants have already been set within the wp-config.php
+ * file and, if so, will populate the options accordingly.
+ *
+ * @return void
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function wds_headless_read_wp_config() {
+	if ( ! _wds_headless_has_defined_all_constants() ) {
+		return;
+	}
+
+	// The consetants and corresponding options names supported by wp-config.php.
+	$config_values = [
+		'HEADLESS_FRONTEND_URL'       => 'wds-headless-frontend-url',
+		'WORDPRESS_PREVIEW_SECRET'    => 'wds-headless-preview-secret',
+		'GRAPHQL_JWT_AUTH_SECRET_KEY' => 'wds-headless-jwt-auth-key',
+	];
+
+	foreach ( $config_values as $constant => $option_name ) {
+		_wds_headless_update_options( $constant, $option_name );
+	}
+}
+
+add_action( 'admin_menu', 'wds_headless_options_page' );
+/**
+ * Adds the plugin menu item to the 'Settings' menu in the WordPress dashboard.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function wds_headless_options_page() {
+	add_options_page(
+		__( 'WDS Headless Theme Settings', 'wds' ),
```

Comment:
```suggestion
		esc_html__( 'WDS Headless Theme Settings', 'wds' ),
```

Please escape all translation string functions.